### PR TITLE
Feature ::  Amazon Athena plugin

### LINKED ISSUE #2937: Amazon Athena
### Is your proposal related to a problem?

Can't query Amazon Athena data.

### Describe the solution you'd like

I'd like to be able to query Athena the same way I can query other Amazon services.

### Describe alternatives you've considered

Custom JS or API integration.

### If the feature is approved, would you be willing to submit a PR?

Not in the near future.

### PR DESCRIPTION
**Pull request by stepinfwd**
closes #2937

package used [https://www.npmjs.com/package/athena-express](url)

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
image

Probably an app-wide issue not related to this PR.

File: plugins/package-lock.json [hunk 1]
```diff
@@ -35,7 +36,6 @@
         "@tooljet-plugins/restapi": "file:packages/restapi",
         "@tooljet-plugins/rethinkdb": "file:packages/rethinkdb",
         "@tooljet-plugins/s3": "file:packages/s3",
-        "@tooljet-plugins/saphana": "file:packages/saphana",
```
Comment (gsmithun4) [inline plugins/package-lock.json:38]: Saphana is removed, please add it

File: plugins/packages/athena/lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      await athenaClient.query('SHOW TABLES');
+    } catch (error) {
+      throw new Error(error);
+    }
+    return {
+      status: 'ok',
+    };
+  }
+  async getConnection(sourceOptions: SourceOptions): Promise<any> {
+    const AthenaExpress = require('athena-express'),
```
Comment (gsmithun4) [inline plugins/packages/athena/lib/index.ts:41]: shall we import them at the top?

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Comment by stepinfwd** [comment]
All changes are updated @gsmithun4

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Comment by Nandinisaha13** [comment]
<html>
<body>
<google-sheets-html-origin>Test Cases

Test Case | Steps | Result
-- | -- | --
Should be able to view the icon for amazon athena in light mode. | Create an application. | Application gets created.
  | Click on add data source. | The datasource pop up opens up.
  | Search amazon athena. | The icon is visible.
Should be able to view the icon for amazon athena in dark mode. | Create an application. | Application gets created.
  | Click on dark mode. | The application gets into dark mode.
  | Click on add data source. | The datasource pop up opens up.
  | Search amazon athena. | The icon is visible.
Should be able to get results for queries. | Create an application. | Application gets created.
  | Click on add data source. | The datasource pop up opens up.
  | Click on athena. | Athena connection pop up is displayed.
  | Enter the credentials. | The credentials are entered.
  | Click on test connection. | The connection is verified.
  | Click on save. | The datasource is saved.
  | Create a query for creating database & click preview. | Database is created.
  | Create another query for displaying the database created & click preview. | Created database name's are displayed.
Should be able to get results for next token & query execution id. | Create an application. | Application gets created.
  | Click on add data source. | The datasource pop up opens up.
  | Click on athena. | Athena connection pop up is displayed.
  | Enter the credentials. | The credentials are entered.
  | Click on test connection. | The connection is verified.
  | Click on save. | The datasource is saved.
  | Create a query for limiting students with some pagination value less than 10 & click on preview. | The query displayed the correct results.


</body>
</html>

File: plugins/packages/athena/lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    return {
+      status: 'ok',
+      data: result,
+    };
+  }
+
+  async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
+    const athenaClient = await this.getConnection(sourceOptions);
+    try {
+      await athenaClient.query('SHOW TABLES');
```
Comment (gsmithun4) [inline plugins/packages/athena/lib/index.ts:34]: why are we using try catch here? it doesn't make any difference since we are throwing the same error

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Pull request merged**